enforcer-rules/src/test/java/org/apache/maven/enforcer/rules/files/TestRequireFilesExist.java

```diff
@@ -50,12 +56,34 @@ void testFileExists() throws Exception {
     }
 
     @Test
-    void testFileOsIndependentExists() {
-        rule.setFilesList(Collections.singletonList(new File("POM.xml")));
+    void testSymbolicLinkExists() throws Exception {
+        File canonicalFile = File.createTempFile("canonical_", null, temporaryFolder);
+        File linkFile = Files.createSymbolicLink(
+                        Paths.get(temporaryFolder.getAbsolutePath(), "symbolic.link"),
+                        Paths.get(canonicalFile.getAbsolutePath()))
+                .toFile();
 
-        EnforcerRuleException e = assertThrows(EnforcerRuleException.class, () -> rule.execute());
+        rule.setFilesList(Collections.singletonList(linkFile));
 
-        assertNotNull(e.getMessage());
+        rule.execute();
+    }
+
+    @Test
+    void testSymbolicLinkTargetDoesNotExist() throws Exception {
+        File canonicalFile = File.createTempFile("canonical_", null, temporaryFolder);
+        File linkFile = Files.createSymbolicLink(
+                        Paths.get(temporaryFolder.getAbsolutePath(), "symbolic.link"),
+                        Paths.get(canonicalFile.getAbsolutePath()))
+                .toFile();
+        canonicalFile.delete();
+        rule.setFilesList(Collections.singletonList(linkFile));
+
+        try {
+            rule.execute();
+            fail("Should have received an exception");
+        } catch (Exception e) {
```

Comment:
which exception should be thrown here? Be specific.

Reply:
Done